src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -497,9 +498,16 @@ public IActionResult GetHistory([FromQuery] WalletHistoryRequest request)
                         }
                     }
 
+                    // Sort and filter the history items.
+                    List<TransactionItemModel> itemsToInclude = transactionItems.OrderByDescending(t => t.Timestamp)
+                        .Where(x => string.IsNullOrEmpty(request.SearchQuery) || (x.Id.ToString() == request.SearchQuery || x.ToAddress == request.SearchQuery || x.Payments.Any(p => p.DestinationAddress == request.SearchQuery)))
```

Comment:
What if SearchQuery is null or empty?

Reply:
Ah, that's the beauty of this statement!
The goal is to avoid evaluating `x.Id.ToString() == request.SearchQuery` (for example) if SearchQuery is empty.
So:
if SearchQuery is null or empty then the evaluation stops as the statement returns `true`.
if SearchQuery is not null or empty, then the evaluation moves on to the rest of the statement.